dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/EndpointManager.java

```diff
@@ -77,13 +74,21 @@ public synchronized void addSession(Endpoint ep, SessionExt sess) {
         // put the updated list back into the Map
         endpointSessionMap.put(cl, sa);
 
-        if (tc.isDebugEnabled()) {
-            Tr.debug(tc, "added session of: " + sess + "  to endpoint class of: " + cl + " using list of: " + sa + " in endpointmanager of: " + this);
-        }
+         String id = sess.getSessionImpl().getHttpSessionID();
 
-        String id = sess.getSessionImpl().getHttpSessionID();
         if (id != null) {
-            httpSessionMap.put(id, sess);
+            ArrayList<Session> http_sa = httpSessionMap.get(id);
```

Comment:
A bit confusing are these httpSessions or wsoc sessions?

Reply:
We only import `javax.websocket.Session`, so the Sessions are websocket. 

Only part of the HttpSesssion used in this class is the Id.